### PR TITLE
Redefine getMode().test to only be true in local development

### PR DESCRIPTION
**Pull request by jridgewell**
Right now in production, we still include all of the code guarded by `getMode().test`. That's because it's a configuration value and not a compile time constant.

Now, test will be stripped out when running `gulp dist` if you do not include `--fortesting`. And other related `getMode` cleanups.

### CODE REVIEW

File: build-system/eslint-rules/get-mode-usage.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+'use strict';
+
+module.exports = function (context) {
+  return {
+    'CallExpression[callee.name=getMode]': function (node) {
+      if (
+        node.parent.type === 'MemberExpression' &&
+        node.parent.object === node
```
Comment (choumx) [inline build-system/eslint-rules/get-mode-usage.js:23]: Can you add an example of this in a comment?

File: build-system/babel-config/pre-closure-config.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ function getPreClosureConfig() {
     !isCheckTypes
       ? './build-system/babel-plugins/babel-plugin-transform-json-configuration'
       : null,
-    argv.esm
+    !(argv.fortesting || isCheckTypes)
```
Comment (choumx) [inline build-system/babel-config/pre-closure-config.js:86]: Why this change?

File: src/mode-object.js [hunk 3]
```diff
@@ -27,7 +27,6 @@ export function getModeObject(opt_win) {
     localDev: getMode(opt_win).localDev,
     development: getMode(opt_win).development,
     esm: IS_ESM,
-    filter: getMode(opt_win).filter,
```
Comment (choumx) [inline src/mode-object.js:30]: Unused?

File: src/mode.js [hunk 4]
```diff
@@ -71,10 +71,9 @@ function getMode_(win) {
   const IS_DEV = true;
   const IS_MINIFIED = false;
 
-  const localDevEnabled = !!AMP_CONFIG.localDev;
   const runningTests =
-    !!AMP_CONFIG.test || (IS_DEV && !!(win.__AMP_TEST || win.__karma__));
-  const isLocalDev = IS_DEV && (localDevEnabled || runningTests);
+    IS_DEV && !!(AMP_CONFIG.test || win.__AMP_TEST || win.__karma__);
```
Comment (choumx) [inline src/mode.js:75]: I don't think all of our `gulp` test commands set `IS_DEV`.

**Comment by choumx** [comment]
> Right now in production, we still include all of the code guarded by getMode().test. 

Nice find!

File: build-system/babel-config/pre-closure-config.js (hunk 2 above)
Comment (rsimha) [reply build-system/babel-config/pre-closure-config.js:86]: This was an intentional change that I remember making. Without it, it was impossible to get `v0.js` to be byte-for-byte identical before and after #27683. See https://github.com/ampproject/amphtml/pull/27683#issuecomment-611822634

Having said that, I'm going to defer to your knowledge of `babel` transforms and when they should be run. Can you confirm that this doesn't affect the minified output in undesirable ways?

**Comment by jridgewell** [comment]
```
dist/v0.js: Δ -0.17KB
```
